tony-history-server/app/utils/HdfsUtils.java

```diff
@@ -63,27 +59,52 @@ static String contentOfHdfsFile(FileSystem fs, Path filePath) {
   }
 
   /**
-   * List all metadata file paths in {@code histFolder}.
+   * Extract the job id portion from the file path string.
+   * @param path file path string.
+   * @return job id
+   */
+  public static String getJobId(String path) {
+    if (path.length() == 0) return "";
```

Comment:
This check is redundant as the following lines do the same thing.

I think you should add a check (and a test) for `path == null` though.

Finally, please use braces, even for one-line `if` bodies, i.e.:
```
if (...) {
  ...
}
```

Reply:
Right. Fixed: https://github.com/linkedin/TonY/pull/81/commits/8b8308d29c6be02cdcafb7a2e32a4a7b0d028316.